prowler/providers/aws/services/rds/rds_instance_no_public_access/rds_instance_no_public_access.py

```diff
@@ -17,18 +18,25 @@ def execute(self):
                 f"RDS Instance {db_instance.id} is not publicly accessible."
             )
             if db_instance.public:
+                report.status_extended = f"RDS Instance {db_instance.id} is set as publicly accessible, but is not publicly exposed."
                 # Check if any DB Instance Security Group is publicly open
                 if db_instance.security_groups:
                     report.status = "PASS"
-                    report.status_extended = f"RDS Instance {db_instance.id} is public but filtered with security groups."
-                    for security_group in ec2_client.security_groups:
-                        if (
-                            security_group.id in db_instance.security_groups
-                            and security_group.public_ports
-                        ):
-                            report.status = "FAIL"
-                            report.status_extended = f"RDS Instance {db_instance.id} is set as publicly accessible."
-                            break
+                    report.status_extended = f"RDS Instance {db_instance.id} is set as publicly accessible but filtered with security groups."
+                    db_instance_port = db_instance.endpoint.get("Port")
+                    if db_instance_port:
+                        for security_group in ec2_client.security_groups:
+                            if security_group.id in db_instance.security_groups:
+                                for ingress_rule in security_group.ingress_rules:
+                                    if check_security_group(
+                                        ingress_rule,
+                                        "tcp",
+                                        [db_instance_port],
+                                        any_address=True,
```

Comment:
I think we should do this in the RDS service checking if the RDS instance port is open to the internet. The check should only verify that.

Reply:
I would prefer only importing the EC2 service if this check is executed.

Reply:
but it is already instantiated there, I think makes more sense to have that logic in the service since you want to store that attribute.